### PR TITLE
stm32wl: Fix gpio irq handling

### LINKED ISSUE #42218: stm32wl: Issue when disabling gpio interrupt
**Describe the bug**
`tests/drivers/gpio/gpio_api_1pin` is failed on nucleo_wl55jc
cf https://github.com/zephyrproject-rtos/test_results/pull/775

On some gpio lines it could be impossible to disable IRQ since one exti function is not able to report the correct IRQ source.

**To Reproduce**
1. west build -p -b nucleo_wl55jc tests/drivers/gpio/gpio_api_1pin
2. west flash

**Expected behavior**
Should be passed

**Impact**
Abnormal behavior on GPIO call backs for some specific GPIO lines

### PR DESCRIPTION
**Pull request by erwango**
EXTI LL API was buggy and was not returning correct source input for some GPIO lines.

- Fix this 
- Add an overlay to enable gpio_basic_api test on nucleo_wl55jc
- Fix openocd configuration

Fixes #42218

### CODE REVIEW

File: dts/arm/st/wl/stm32wl.dtsi [hunk 1]
```diff
@@ -105,11 +105,11 @@
 			label = "STM32_CLK_RCC";
 		};
 
-		exti: interrupt-controller@58000800 {
+		exti: interrupt-controller@40010000 {
```
Comment (FRASTM) [inline dts/arm/st/wl/stm32wl.dtsi:108]: There are 2 peripherals : EXTI and SYSCFG.

In the /drivers/interrupt_controller/intc_exti_stm32.c,  the **EXTI peripheral**   is actually  registered @ 0x58000800 when used by the `stm32_exti_enable` or `stm32_exti_disable`  functions.
-->  LL_EXTI_EnableIT_0_31() etc  refers to the EXTI peripheral.


This differs for stm32wl (and others stm32 mcus like stm32wb) from the **SYSCFG peripheral**  which is registered @  0x40010000 and  configured  by the   gpio_stm32_set_exti_source function (in the  drivers/gpio/gpio_stm32.c)
-->  the LL_SYSCFG_SetEXTISource(port, line) refers to the SYSCFG peripheral.

**Review by FRASTM — COMMENTED** [review]
check of the exti periph register bank (comparing to stm32wb)

File: dts/arm/st/wl/stm32wl.dtsi (hunk 1 above)
Comment (erwango) [reply dts/arm/st/wl/stm32wl.dtsi:108]: Right, I mixed both in the debug session. Will fix, txs

**Review by FRASTM — APPROVED** [review]

**Review by mnkp — APPROVED** [review]
LGTM. Though, @erwango if you have time you could update the commit message of "boards: nucleo_wl55jc: Fix openocd configuration" commit and explain shortly why the current configuration was wrong.

**Comment by erwango** [comment]
> LGTM. Though, @erwango if you have time you could update the commit message of "boards: nucleo_wl55jc: Fix openocd configuration" commit and explain shortly why the current configuration was wrong.

This is right. I've reworked commit to provide more clarity and  also added a note in board documentation.

**Review by ABOSTM — APPROVED** [review]

**Review by MaureenHelm — APPROVED** [review]

**Pull request merged**